src/lib/y2network/sysconfig/config_writer.rb

```diff
@@ -36,33 +36,39 @@ module Sysconfig
     class ConfigWriter
       include Yast::Logger
 
+      SECTIONS = [:routing, :drivers, :interfaces, :connections, :dns, :hostname].freeze
+
       # Writes the configuration into YaST network related modules
       #
       # @param config     [Y2Network::Config] Configuration to write
       # @param old_config [Y2Network::Config] Old configuration
-      def write(config, old_config = nil)
+      def write(config, old_config = nil, sections: :all)
+        sections = SECTIONS if sections == :all
         log.info "Writing configuration: #{config.inspect}"
         log.info "Old configuration: #{old_config.inspect}"
-        write_ip_forwarding(config.routing) if config.routing
-        write_interface_changes(config, old_config)
-
-        # update /etc/sysconfig/network/routes file
-        file = routes_file_for(nil)
-        file.routes = find_routes_for(nil, config.routing.routes)
-        file.save
+        write_ip_forwarding(config.routing) if sections.include?(:routing)
+        write_interface_changes(config, old_config) if sections.include?(:interfaces)
 
-        write_drivers(config.drivers)
-        write_interfaces(config.interfaces)
-        write_connections(config.connections, old_config)
-        write_dns_settings(config, old_config)
-        write_hostname_settings(config, old_config)
+        write_routes(config.routing.routes) if sections.include?(:routing)
+        write_drivers(config.drivers) if sections.include?(:drivers)
+        write_interfaces(config.interfaces) if sections.include?(:interfaces)
+        write_connections(config.connections, old_config) if sections.include?(:connections)
+        write_dns_settings(config, old_config) if sections.include?(:dns)
+        write_hostname_settings(config, old_config) if sections.include?(:hostname)
```

Comment:
well, when I see this I think it deserve a bit of cleaning and unification. So what is idea? Always has same arguments `config, old_config` and method pick what it needs. and match method name with section name.
And then this method can look like 
```
# add note that sections order is important or not?
SECTIONS.each do { |s| send(:"write_#{s}", config, old_config) if sections.include?(s) }
```
this way adding new section write is simple putting it in SECTIONS and then write write_<section> method.